index.d.ts

```diff
@@ -0,0 +1,5 @@
+export * from './advanced';
+
+import { QueryMiddleware } from './advanced';
```

Comment:
Why is this line needed? Query middleware is exported from advanced already? Unless I'm confused.

Reply:
`export * from` re-exports all the types from the `advanced` module -- treating the `advanced` module as if it were the whole project (to keep type definitions in one file).

The only difference between `index` and `advanced` as far as I can see is that `index` exposes a default `queryMiddleware` (with superagent) of type `QueryMiddleware`. Hence, this type is used in the declaration right below -- saying that this module contains a constant of type `QueryMiddleware`

Reply:
Oh I see. Make sense.